Tests/Functional/Controller/HeadlessWebsiteControllerTest.php

```diff
@@ -37,12 +43,19 @@ public static function setUpBeforeClass(): void
         ]);
     }
 
+    protected function setUp(): void
+    {
+        static::ensureKernelShutdown();
```

Comment:
IMO it makes more sense to ensure that the kernel is shut down **before** each test, because a test will fail, if it tries to boot an already booted kernel.

Reply:
We should only do it where really needed. This is from me really confusing seeing ensureKernelShutdown as first of the setUp function.

Reply:
fixed